program-runtime/src/invoke_context.rs

```diff
@@ -1077,66 +1076,64 @@ impl<'a> InvokeContext<'a> {
             .get_key_of_account_at_index(index_in_transaction)
     }
 
-    /// Set the original account lengths
-    pub fn set_orig_account_lengths(
+    // Set this instruction syscall context
+    pub fn set_syscall_context(
         &mut self,
+        check_aligned: bool,
+        check_size: bool,
         orig_account_lengths: Vec<usize>,
+        allocator: Rc<RefCell<dyn Alloc>>,
     ) -> Result<(), InstructionError> {
         *self
-            .orig_account_lengths
+            .syscall_context
             .last_mut()
-            .ok_or(InstructionError::CallDepth)? = Some(orig_account_lengths);
+            .ok_or(InstructionError::CallDepth)? = Some(SyscallContext {
+            check_aligned,
+            check_size,
+            orig_account_lengths,
+            allocator,
+        });
         Ok(())
     }
 
-    /// Get the original account lengths
-    pub fn get_orig_account_lengths(&self) -> Result<&[usize], InstructionError> {
-        self.orig_account_lengths
-            .last()
-            .and_then(|orig_account_lengths| orig_account_lengths.as_ref())
-            .map(|orig_account_lengths| orig_account_lengths.as_slice())
-            .ok_or(InstructionError::CallDepth)
-    }
-
-    // Set should alignment be enforced during user pointer translation
-    pub fn set_check_aligned(&mut self, check_aligned: bool) {
-        self.check_aligned = check_aligned;
-    }
-
     // Should alignment be enforced during user pointer translation
     pub fn get_check_aligned(&self) -> bool {
```

Comment:
How about replacing `get_check_aligned()`, `get_check_size()`, `get_orig_account_lengths()` and `get_allocator()` by one accessor: `InvokeContext::get_current_syscall_context()`, analogous to `TransactionContext::get_current_instruction_context()`?

That way all these helper functions could be replaced by direct access to the struct properties.

Reply:
Maybe later, exposing allocators and orig account lenghs to get the alignment check seems, eh